pycozmo/conn.py

```diff
@@ -259,9 +304,9 @@ def connect(self) -> None:
         except InterruptedError:
             pass
 
-    def send(self, pkt: Packet) -> None:
+    def send(self, pkt: Packet, repeat=False) -> None:
```

Comment:
`repeat` does not need to be exposed to the application layer.

Cozmo is using a form of the [selective-repeat ARQ protocol](https://en.wikipedia.org/wiki/Selective_Repeat_ARQ) . It guarantees to the application layer that a sequence of packets that are sent, will be received on the other side in the same order and without duplicates.

Reply:
Do we have this already implemented? I'm sorry, I'm a bit clueless about network protocols so I probably missed a ton of stuff that is already there.
It makes a lot of sense that cozmo uses something like that, but I am also wondering, how does cozmo acknowledge that a message is received? I thought first that maybe not all frames are confirmed, which would explain why I saw that relation between the audio and the Keyframe packets, but now looking at this, it would make more sense that in all responses from cozmo there would be an acknowledgement of the received frames, does this happen? Then maybe the Keyframe is a completely unrelated thing. Tbh that would make things much easier for the whole communication. I don't really like the idea of having different behaviours for different packets.